lib/integrations/awsoidc/tags.go

```diff
@@ -79,3 +80,16 @@ func (d awsTags) MatchesECSTags(resourceTags []ecsTypes.Tag) bool {
 
 	return true
 }
+
+// ToIAMTags returns the default tags using the expected type for IAM resources: [iamTypes.Tag]
+func (d awsTags) ToIAMTags() []iamTypes.Tag {
+	iamTags := make([]iamTypes.Tag, 0, len(d))
+	for k, v := range d {
+		k, v := k, v
```

Comment:
Key and Value of `d awsTags[map[string]string` are "value" types. Why do we need make a second copy ?

Reply:
But then we use their address because `iamTypes.Tag.Key/Value` require string pointers.
I just removed the second copy and the tests in `tags_test.go` fail because of it

Reply:
i think they will fix this in newer go
https://github.com/golang/go/wiki/LoopvarExperiment.
An alternative is to use `aws.String()` instead of `&k/&v`